File: rest-common/src/main/java/org/chronopolis/rest/ZonedDateTimeConverter.java
package org.chronopolis.rest;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;
import java.sql.Timestamp;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;

/**
 * Simple converter for using a ZonedDate
 *
 * Created by shake on 3/21/16.
 */
@Converter
public class ZonedDateTimeConverter implements AttributeConverter<ZonedDateTime, Timestamp> {

    @Override
    public Timestamp convertToDatabaseColumn(ZonedDateTime dateTime) {
        if (dateTime != null) {
            return Timestamp.valueOf(dateTime.toLocalDateTime());
        }

        return null;
    }

    @Override
    public ZonedDateTime convertToEntityAttribute(Timestamp timestamp) {
        if (timestamp != null) {
            return timestamp.toLocalDateTime().atZone(ZoneOffset.UTC);
        }

        return null;
    }
}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/kot/entities/converters/ZonedDateTimeConverter.kt
package org.chronopolis.rest.kot.entities.converters

import java.sql.Timestamp
import java.time.LocalDateTime
import java.time.ZoneOffset
import java.time.ZonedDateTime
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter
class ZonedDateTimeConverter : AttributeConverter<ZonedDateTime, Timestamp> {

    override fun convertToDatabaseColumn(dateTime: ZonedDateTime?): Timestamp {
        if (dateTime != null) {
            return Timestamp.valueOf(dateTime.toLocalDateTime())
        }

        // not sure if this is a good idea or not
        return Timestamp.valueOf(LocalDateTime.now())
    }

    override fun convertToEntityAttribute(timestamp: Timestamp?): ZonedDateTime {
        if (timestamp != null) {
            return timestamp.toLocalDateTime().atZone(ZoneOffset.UTC)
        }

        return ZonedDateTime.now()
    }
}